sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala

```diff
@@ -107,14 +106,27 @@ case class CustomShuffleReaderExec private(
       (numPartitionsMetric.id, partitionSpecs.length.toLong)
 
     if (hasSkewedPartition) {
-      val skewedMetric = metrics("numSkewedPartitions")
-      val numSkewedPartitions = partitionSpecs.collect {
-        case p: PartialReducerPartitionSpec => p.reducerIndex
-      }.distinct.length
-      skewedMetric.set(numSkewedPartitions)
-      driverAccumUpdates = driverAccumUpdates :+ (skewedMetric.id, numSkewedPartitions.toLong)
-    }
+      val skewedPartitions = metrics("numSkewedPartitions")
+      val skewedSplits = metrics("numSkewedSplits")
+
+      val skewedMetrics = new mutable.HashMap[Int, Long]()
```

Comment:
My old comment still applies: https://github.com/apache/spark/pull/28109#discussion_r402900400

Reply:
Here we need to calculate the number of splits for every skewed partition not only the total number. We may need this `HashMap`